src/persist/src/postgres.rs

```diff
@@ -340,23 +340,17 @@ impl Consensus for PostgresConsensus {
         }
 
         let result = if let Some(expected) = expected {
-            // Only insert the new row if:
-            // - sequence number expected is already present
-            // - expected corresponds to the most recent sequence number
-            //   i.e. there is no other sequence number > expected already
-            //   present.
-            //
-            // This query has also been written to execute within a single
-            // network round-trip (instead of a slightly simpler implementation
-            // that would call `BEGIN` and have multiple `SELECT` queries).
-            let q = "INSERT INTO consensus SELECT $1, $2, $3 WHERE
-                     EXISTS (
-                        SELECT * FROM consensus WHERE shard = $1 AND sequence_number = $4
-                     )
-                     AND NOT EXISTS (
-                         SELECT * FROM consensus WHERE shard = $1 AND sequence_number > $4
-                     )
-                     ON CONFLICT DO NOTHING";
+            // This query is tuned for performance against CockroacbDB.
+            // The insert is designed to hit the fast-path 1-phase commit
+            // of CRDB. Any changes to this query should ensure an
+            // EXPLAIN ANALYZE (VERBOSE) query plan contains `auto commit`
```

Comment:
I'd like to keep the `This query has also been written to execute within a single network round-trip` comment